src/test/java/eu/yals/test/utils/elements/YalsElement.java

```diff
@@ -5,89 +5,89 @@
 import org.openqa.selenium.WebElement;
 
 /**
- * Test methods for Selenide Elements
+ * Test methods for Selenide Elements.
  *
  * @since 2.7
  */
 public class YalsElement {
-  private final WebElement webElement;
+    private final WebElement webElement;
 
-  public static YalsElement wrap(WebElement element) {
-    return new YalsElement(element);
-  }
+    public static YalsElement wrap(WebElement element) {
+        return new YalsElement(element);
+    }
+
+    protected YalsElement(final WebElement element) {
+        this.webElement = element;
+    }
+
+    public void shouldHaveAttr(final String attributeName) {
+        shouldHaveAttr(attributeName, "");
+    }
+
+    public void shouldHaveAttr(final String attributeName, final String exceptedValue) {
+        Assert.assertNotNull("No such element found", webElement);
 
-  protected YalsElement(WebElement element) {
-    this.webElement = element;
-  }
+        String attributeValue = webElement.getAttribute(attributeName);
+        Assert.assertNotNull(String.format("No such attribute '%s'", attributeName), attributeValue);
 
-  public void shouldHaveAttr(String attributeName) {
-    shouldHaveAttr(attributeName, "");
-  }
+        if (StringUtils.isNotBlank(exceptedValue)) {
+            Assert.assertTrue("Attribute is empty", StringUtils.isNotBlank(attributeValue));
+            Assert.assertEquals(exceptedValue, attributeValue);
+        }
+    }
+
+    public void shouldExist() {
+        Assert.assertNotNull("No such element found", webElement);
+    }
 
-  public void shouldHaveAttr(String attributeName, String exceptedValue) {
-    Assert.assertNotNull("No such element found", webElement);
+    public void shouldBeDisplayed() {
+        shouldExist();
+        Assert.assertTrue("Element is not displayed", webElement.isDisplayed());
+    }
+
+    public void shouldNotBeDisplayed() {
+        shouldExist();
+        Assert.assertFalse("Element is displayed", webElement.isDisplayed());
+    }
 
-    String attributeValue = webElement.getAttribute(attributeName);
-    Assert.assertNotNull(String.format("No such attribute '%s'", attributeName), attributeValue);
+    public void shouldBeEnabled() {
+        shouldExist();
+        Assert.assertTrue("Element is disabled", webElement.isEnabled());
+    }
+
+    public void textHas(String phrase) {
+        shouldExist();
+        Assert.assertTrue(
+                String.format("Text: '%s' has no '%s'", webElement.getText(), phrase),
+                webElement.getText().contains(phrase));
+    }
+
+    public void shouldNotBeEmpty() {
+        shouldExist();
+        Assert.assertTrue("Text is empty", StringUtils.isNotBlank(webElement.getText()));
+    }
+
+    public void shouldBeEmpty() {
+        shouldExist();
+        Assert.assertTrue("Text is not empty", StringUtils.isBlank(webElement.getText()));
+    }
+
+    public void shouldHaveText(String text) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter text should be final.](https://app.codacy.com/gh/yadevee/yals/pullRequest?prid=5409395)